lib/rules/selector-not-notation/__tests__/index.mjs

```diff
@@ -140,6 +152,81 @@ testRule({
 			endLine: 1,
 			endColumn: 12,
 		},
+		{
+			code: stripIndent`
+				/* a comment */
+				:not(a, ), /* a comment */
+				:not(a, ),
+				/* a comment */
+				:not(a, )
+				{}
+			`,
+			fixed: stripIndent`
+				/* a comment */
+				:not(a), /* a comment */
+				:not(a),
+				/* a comment */
+				:not(a)
+				{}
+			`,
+			warnings: [
+				{
+					message: messages.expected('simple'),
+					line: 2,
+					column: 1,
+					endLine: 2,
+					endColumn: 10,
+				},
+				{
+					message: messages.expected('simple'),
+					line: 3,
+					column: 1,
+					endLine: 3,
+					endColumn: 10,
+				},
+				{
+					message: messages.expected('simple'),
+					line: 5,
+					column: 1,
+					endLine: 5,
+					endColumn: 10,
+				},
+			],
+		},
+		{
+			code: stripIndent`
+				/* a comment */
+				:not(a, ), /* stylelint-disable-next-line selector-not-notation */
+				:not(b, ),
+				/* a comment */
+				:not(c, )
+				{}
+			`,
+			fixed: stripIndent`
+				/* a comment */
+				:not(a), /* stylelint-disable-next-line selector-not-notation */
+				:not(b,),
```

Comment:
It removes spaces even for ignore cases. 
`selectorParser` appears to strip them out, even with [`{lossless: true}`](https://github.com/postcss/postcss-selector-parser/blob/6158750aab0aed3046a5920bd58e0e0266a4ada2/API.md#processoroptions) set.
https://github.com/stylelint/stylelint/blob/eb20f640b5949701cbf4c963db6dd34d28e081d3/lib/utils/parseSelector.mjs#L20

Reply:
This is interesting!

Please consider opening an upstream issue about this in https://github.com/postcss/postcss-selector-parser if you have time.

Reply:
Just opened an upstream issue.
https://github.com/postcss/postcss-selector-parser/issues/298